### DISCUSSION TITLE
can't seem to get language locales working (ie: 'en' works, but 'zh-hant' does not)

### DISCUSSION THREAD

**Discussion by ivar** [i18next/next-i18next#2125] [Q&A] [answered]
Hey team, 
I'm pretty new to this but I've run into an issue where for some reason locales that have a subvariant are not being loaded correctly.  

My next-i18next.config.js looks like:
```json
  i18n: {
    locales: ["en", "zh-hant", "zh-hans", "moo"],
    defaultLocale: "en",
  }
```
and my folder structure looks like:
```
public/
  locales/
    en/
      common.json
    zh-hant/
      common.json
    zh-hans/
      common.json
    moo/
      common.json
```

I can switch locales in the webapp (using route based locales) and the props value is correct, but the `initialI18nStore` inside `props` is an empty object when I use zh-hant and zh-hans. While it's great that `en` and `moo` work (because we want cows to use the site comfortably) I'm stuck as to why the Chinese variants are failing.  The format of `common.json` is correct in all locale directories.

console logs from my two attempts
unsuccessful:
```
current locale: "zh-hant"
initialI18nStore from props:
{
  'zh-hant': { common: {} },
  en: {
    common: {
      'news.title': 'News English - yo!'
    }
  }
}
```
and successful:
```
current locale: "moo"
initialI18nStore from props:
{
  moo: {
    common: {
      'news.title': 'News English - MOOOOOOOOOOOOOOOOOOOOOOOOOOOO!'
    }
  },
  en: {
    common: {
      'news.title': 'News English - yo!'
    }
  }
}
```
Any help very much appreciated!

**Comment by ivar** [comment]
Ok weird! While digging through discussions here I saw how to enable debugging and found some very interesting logs.
```
current locale: "zh-hant"
i18next::backendConnector: loading namespace common for language zh-Hant failed [Error: ENOENT: no such file or directory, open '/home/ivar/Work/ccmv2/public/locales/zh-Hant/common.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/ivar/Work/ccmv2/public/locales/zh-Hant/common.json'
}
i18next::backendConnector: loading namespace common for language zh-Hans failed [Error: ENOENT: no such file or directory, open '/home/ivar/Work/ccmv2/public/locales/zh-Hans/common.json'] {
  errno: -2,
  code: 'ENOENT',
  syscall: 'open',
  path: '/home/ivar/Work/ccmv2/public/locales/zh-Hans/common.json'
}
```

This is a bit weird, in that nowhere do I specify 'zh-Hant' in my code or config - I always use lowercase for the entire value. Somehow i18next is uppercasing the first letter of the variant I'm using. I realize the capitalization might be more in accordance with language standards, but I prefer the aesthetics of all lowercase.  It seems odd that the library would enforce this capitalization but allow me to define my own arbitrary language string (in this case, the 'moo' language). Is this expected behaviour?

**Comment by adrai** [comment, marked as answer]
Use the correct language codes and try again: https://www.i18next.com/how-to/faq#how-should-the-language-codes-be-formatted
